src/base/AsyncCallQueue.h

```diff
@@ -22,18 +21,13 @@ class AsyncCallQueue
     static AsyncCallQueue &Instance();
 
     // make this async call when we get a chance
-    void schedule(AsyncCall::Pointer &call);
+    void schedule(const AsyncCallPointer &call) { scheduled.add(call); }
 
     // fire all scheduled calls; returns true if at least one was fired
     bool fire();
 
 private:
-    AsyncCallQueue();
```

Comment:
This private declaration should stay (in some shape or form) because it is meaningful -- we are telling the reader that they should not be creating AsyncCallQueue objects (because AsyncCallQueue is designed as a singleton). We can make that idea more explicit if you think it was not explicit enough:

```C++
    // private placement prohibits creation of competing queues by the public
    AsyncCallQueue() = default;
```

Reply:
> AsyncCallQueue is designed as a singleton

No, it was not a singleton because we could still create a copy (via implicit copy constructor). To enforce this pattern I would suggest, in addition, to delete both copy constructor and operator=.

Reply:
>> AsyncCallQueue is designed as a singleton

> No

AsyncCallQueue is designed as a singleton. Whether that design was implemented correctly is a different question ;-).

> we could still create a copy (via implicit copy constructor). To enforce this pattern I would suggest, in addition, to delete both copy constructor and operator=.

Yes, and I would try to go a step further, actually -- since a single AsyncCall object should not be present in two AsyncCallLists, let's try to prohibit duplicating of AsyncCallList objects (moving is OK). If we can do that, then it will automatically prevent duplication of AsyncCallQueues as well (and no code can accidentally _move_ a singleton anyway). I am not sure we can do that because some copying (of empty lists) might be going somewhere already, and we probably do not want to improve that code in this PR.

Long-term, we might even replace the whole AsyncCallQueue class with a pair of global Schedule()/FireScheduled() functions, but assessing that change is out of scope here, of course, and prohibiting AsyncCallList copying is still valuable even if AsyncCallQueue is gone.

While you are at it please also delete an extra empty line at the very end of AsyncCallList class declaration, before the closing curly brace.

Reply:
> I was talking about AsyncCallList objects.

Done (at the second attempt) - I misinterpretted your initial suggestion.